### PR TITLE
Fix #470

### LINKED ISSUE #470: Bug in `entr` call to `watch_folder`
Line 822 of Revise.jl calls `FileWatching.watch_folder`. It then calls `ret.changed` on the result. But `FileWatching.watch_folder` returns a pair:

```julia
julia> FileWatching.watch_folder("src")
"extra.jl" => FileWatching.FileEvent(false, true, false)
```
As the documentation says:
> The returned value is an pair where the first field is the name of the changed file (if available) and the second field is an object with boolean fields changed, renamed, and timedout, giving the event.

Should be an easy fix. I'll make a pull request.

### PR DESCRIPTION
**Pull request by bogiebro**
Line 822 of Revise.jl calls `FileWatching.watch_folder`. It then calls `ret.changed` on the result. But `FileWatching.watch_folder` returns a pair:

```julia
julia> FileWatching.watch_folder("src")
"extra.jl" => FileWatching.FileEvent(false, true, false)
```
As the documentation says:
> The returned value is an pair where the first field is the name of the changed file (if available) and the second field is an object with boolean fields changed, renamed, and timedout, giving the event.

### CODE REVIEW

File: src/Revise.jl [hunk 1]
```diff
@@ -819,9 +819,13 @@ function entr(f::Function, files, modules=nothing; postpone=false, pause=0.02)
...
             postpone || f()
             for file in files
-                waitfor = isdir(file) ? watch_folder : watch_file
+                is_file_dir = isdir(file)
                 @async while active
-                    ret = waitfor(file, 1)
+                    if is_file_dir
+                        ret = watch_folder(file, 1)[2]
+                    else
+                        ret = watch_File(file, 1)
```
Comment (timholy) [inline src/Revise.jl:827]:
```suggestion
                        ret = watch_file(file, 1)
```

**Comment by mlhetland** [comment]
> CC @mlhetland, since he's kind of the de facto expert on `entr`.

Heh, more a fan than an expert, but thanks :-) Seems like a good idea to me. (The [original `entr`](http://eradman.com/entrproject/) also has the ability to watch directories. It requires a specific flag, `-d`, but I see no need for that in our case, really, as we'll be pretty explicit about supplying the directory name, anyway?)

**Comment by bogiebro** [comment]
I updated the docstrings, and I'm happy to add a test too. Is there any way to run a subset of tests in the suite, rather than of all of them (as in Pkg.test)?

**Comment by timholy** [comment]
Sorry for the delay, I have been very distracted lately. Yes, you can run a subset of tests:
- julia 1.3 or higher: `Pkg.test("Revise", test_args=["Method deletion"])` (you can add more than one testset)
- from the shell prompt, if you have all test dependencies in your environment: `julia runtests.jl "Method deletion"`
- from an interactive session: `push!(ARGS, "Method deletion"); include("runtests.jl")`

**Comment by ffevotte** [comment]
Just to tidy things up and let everyone interested know:
- @bogiebro: #470 should be fixed, now that #500 has been merged. Thanks for getting this started!
- @timholy: I think this PR could be closed now

**Pull request closed without merging**